iocage_lib/ioc_json.py

```diff
@@ -206,35 +206,54 @@ def retrieve_cpu_sets():
     def validate_cpuset_prop(value, raise_error=True):
         failed = False
         cpu_sets = IOCCpuset.retrieve_cpu_sets() + 1
+        cpu_set_value_err = False
 
         if not any(
+            cond for cond in (
+                re.findall(
+                    r'^(\d+(,\d+)*)$',
+                    value
+                ),
+                value in ('off', 'all'),
+                re.findall(
+                    r'^(\d+-\d+)$',
+                    value
+                )
+            )
+        ):
+            failed = True
+        elif not any(
             cond for cond in (
                 re.findall(
                     fr'^(?!.*(\b\d+\b).*\b\1\b)'
                     fr'((?:{"|".join(map(str, range(cpu_sets)))})'
                     fr'(,(?:{"|".join(map(str, range(cpu_sets)))}))*)?$',
                     value
                 ),
-                value in ('off', 'all'),
                 re.findall(
                     fr'^(?:{"|".join(map(str, range(cpu_sets - 1)))})-'
                     fr'(?:{"|".join(map(str, range(cpu_sets)))})$',
                     value
                 ) and int(value.split('-')[0]) < int(value.split('-')[1])
             )
         ):
-            failed = True
-
-        if failed and raise_error:
+            cpu_set_value_err = True
+
+        if (failed or cpu_set_value_err) and raise_error:
+            message = 'Please specify a valid format for cpuset ' \
+                      'value.\nFollowing 4 formats are supported:\n' \
+                      '1) comma delimited string i.e 0,1,2,3\n' \
+                      '2) a range of values i.e 0-2\n' \
+                      '3) "all" - all would mean using all cores\n' \
+                      '4) off'
+            if cpu_set_value_err:
+                message = 'Please make sure the provided cpus fall in the ' \
+                          'correct range of cpus.\nValid cpu entries can be ' \
+                          'found by "cpuset -g -s 0"'
```

Comment:
Since we know it's now failed, let's store that command in a variable and show the values to the user, saves them a step

Reply:
@skarekrow this is a good idea ;)